src/services/resource.js

```diff
@@ -210,6 +327,394 @@ module.exports = class resourceHelper {
+				],
+			},
+			reviewer_id: { [Op.notIn]: [loggedInUserId] },
+		}
+		const reviewsResponse = await reviewsQueries.findAll(reviewsFilter, ['resource_id'])
+
+		if (reviewsResponse) {
+			// push resource ids to resourceIdsToBeRemoved array
+			const resourceIdsToBeRemoved = reviewsResponse.map((item) => item.resource_id)
+			return resourceIdsToBeRemoved
+		} else {
+			return []
+		}
+	}
+
+	// this function fetches review levels from the reviews table for
+	// an organization, roles and resouce type and returns this as an object
+	// Ex : this function returns
+	// {
+	// 	project : 1,
+	// 	observation : 4
+	// }
+	static async fetchResourceReviewLevel(organization_id, userRoleTitles, resourceTypeList) {
+		// fetch review levels according to roles in the organization
+		const reviewLevelDetails = await reviewStagesQueries.findAll(
+			{
+				organization_id,
+				role: {
+					[Op.in]: userRoleTitles,
+				},
+				resource_type: {
+					[Op.in]: resourceTypeList,
+				},
+			},
+			{ attributes: ['resource_type', 'level'], order: [['level', 'ASC']] }
+		)
+
+		if (reviewLevelDetails) {
+			// arrange it as a key-value pair for ease of use
+			const resourceWiseLevels = reviewLevelDetails.reduce((acc, item) => {
+				acc[item.resource_type] = item.level
+
+				return acc
+			}, {})
+
+			return resourceWiseLevels
+		}
+		return {}
+	}
+	// fetch the resources types of an organization
+	static async fetchResourceReviewTypes(organization_id) {
+		try {
+			// Fetch organization-based configurations for resources
+			const configList = await configs.list(organization_id)
+
+			// Map resource types to their review types
+			const resourceWiseReviewType = configList.result.resource.reduce((acc, item) => {
+				acc[item.resource_type] = item.review_type
+				return acc
+			}, {})
+
+			let resourceTypesInSequentialReview = []
+			let resourceTypesInParallelReview = []
+
+			for (const [key, value] of Object.entries(resourceWiseReviewType)) {
+				value === common.REVIEW_TYPE_SEQUENTIAL
+					? resourceTypesInSequentialReview.push(key)
+					: resourceTypesInParallelReview.push(key)
+			}
+
+			// Return the categorized resource types
+			return {
+				sequential: resourceTypesInSequentialReview,
+				parallel: resourceTypesInParallelReview,
+			}
+		} catch (error) {
+			throw error
+		}
+	}
+
+	static async fetchReviewersInprogressResources(user_id, uniqueOrganizationIds) {
+		// fetch resources under user id from the org which are review inProgress and changes updated by creator
+		const filter = {
+			reviewer_id: user_id,
+			organization_id: { [Op.in]: uniqueOrganizationIds },
+			status: { [Op.in]: [common.REVIEW_STATUS_INPROGRESS, common.REVIEW_STATUS_CHANGES_UPDATED] },
+		}
+		let result = await reviewsQueries.findAll(filter, ['resource_id', 'organization_id'])
+		return result
+	}
+
+	static async fetchReviewerNotesForResources(reviewer_id = null, resource_ids, organization_ids) {
+		let filter = {
```

Comment:
@adithyadinesh0412 is this required for up for review ?

Reply:
not required for up for review. removed